SPECS-EXTENDED/azl-otel-collector/azl-otel-collector.spec

```diff
@@ -0,0 +1,73 @@
+# Disabled debuginfo package as the autogenerated 'debugfiles.list' file is empty.
+# In other words there were no debug symbols to package.
+%global debug_package %{nil}
+Summary:        Azure Linux OpenTelemetry Collector Distribution
+Name:           azl-otel-collector
+Version:        0.124.0
+Release:        1%{?dist}
+License:        MIT
+Vendor:         Microsoft Corporation
+Distribution:   Azure Linux
+URL:            https://github.com/microsoft/azl-otel-collector
+Source0:        https://azurelinuxsrcstorage.blob.core.windows.net/sources/core/azl-otel-collector-0.124.0-test.tar.gz
+Source1:        azl-otel-collector-0.124.0-test-vendor.tar.gz
```

Comment:
Adit, what does this file do? and come from?

Reply:
this file is created by running `go mod vendor` locally and then creating & uploading a vendor tarball through the publishing pipeline.

Packages with Go dependencies need their vendor folder available during build, so this is the pattern across numerous packages in azl.